src/libxively/sft/xi_sft_logic.c

```diff
@@ -0,0 +1,98 @@
+/* Copyright (c) 2003-2017, LogMeIn, Inc. All rights reserved.
+ *
+ * This is part of the Xively C Client library,
+ * it is licensed under the BSD 3-Clause license.
+ */
+
+#include <xi_sft_logic.h>
+#include <stddef.h>
+#include <xi_macros.h>
+
+xi_state_t xi_sft_make_context( xi_sft_context_t** context,
+                                const char** updateable_files,
+                                uint16_t updateable_files_count,
+                                fn_send_control_message fn_send_message,
```

Comment:
exactly. This is the decoupling. With this move the xi_sft_logic.c does not need to include the xi_control_topic_layer.h. Saying it one level higher: SFT does not know about Control Topic Layer, one level more: SFT does not know about transport layer? Maybe this is true. It just knows the protocol / logic.
I would have done this same way (decoupled) if there isn's a requirement to inject this SFT into other clients. Benefit of decoupling is the ease of unit testing. Thus the SFT can be tested without the control topic :) Lots of good things come from decoupling.